src/noaa_apt.rs

```diff
@@ -380,11 +380,26 @@ pub fn decode(
     encoder.set_depth(png::BitDepth::Eight);
     let mut writer = encoder.write_header()?;
 
-    writer.write_image_data(&signal[..])?;
+    if settings.rotate_image {
+        context.status(0.98, "Rotating output image".to_string());
+        debug!("Rotating image");
+
+        let mut rotated_image = signal.clone();
+        rotated_image.reverse();
```

Comment:
You can probably eliminate the clone here by using iterator:

```rust
let rotated_image = signal.iter().rev();
```

and then on line 392 you continue with the same iterator:

```rust
let first_part = rotated_image.take(signal.len() - (PX_PER_ROW / 2) as usize));
let fixed_image: Signal = rotated_image.chain(first_part).copied().collect();
```

didn't test this, but probably should work something like that.

Reply:
@oherrala, thanks! I did use `.iter().rev()` in this new revision thanks to your suggestion.